src/query/execute_schema.c

```diff
@@ -6952,6 +6971,40 @@ do_promote_partition_by_name (const char *class_name, const char *part_num, char
   return NO_ERROR;
 }
 
+/*
+ * has_notnull_unique_constraints() - Check if attribute is NOT NULL and UNIQUE
+ *    return: true if the attribute has both NOT NULL and UNIQUE constraints,
+ *            false otherwise
+ *    smattr(in): attribute to check
+ *
+ * Note:
+ *    The decision is based on both smattr->flags (for NOT NULL)
+ *    and smattr->constraints (for UNIQUE).
+ *    - If the attribute has both NOT NULL and UNIQUE constraints, return true
+ *      so that the UNIQUE flags are preserved.
+ *    - Otherwise, return false and the UNIQUE-related flags will be reset.
+ */
+static bool
+has_notnull_unique_constraints (const SM_ATTRIBUTE * smattr)
```

Comment:
has_notnull_unique_constraints() 함수명에 대해 두번째 제안드립니다. 사유는
1. 이 함수는 partition table의 제약사항 때문에 문제 없이 동작합니다. (이 제약들은 커멘트로 남겨주시면, SM_ATTRIBUTE를 개별적으로 검사해도 되는 이유가 명확해 질 것 같습니다.)
    - pk 또는 uk는 반드시 partition key여야 한다
    - partition key는 단일 컬럼만 허용 된다.
2. 1번의 제약사항이 없다면, 복합키에 걸린 제약을 검사해야 하기 때문에 SM_ATTRIBUTE 수준에서 판단할 수 없습니다.
3. 따라서, 이 함수가 파티션 테이블에 국한하여 사용되는 함수임을 명확하게 하기 위해 제안드립니다. 검토 후 의견주세요.
```suggestion
is_part_key_pk_or_uk_notnull (const SM_ATTRIBUTE * smattr)
```

Reply:
@hornetmj 위에서 언급한 함수의 이름과 동작이 정리되면 다시 검토해보려합니다.

Reply:
@youngjun9072 이 부분에 대해서도 검토 요청드립니다.

Reply:
@hornetmj 변수명을 has_notnull_unique 로 유지하기로하였고, 함수의 내용도 PK판별을 하지 않고 not null unique에 대해서만 수행중(PK는 언제나 상속합니다)이기때문에 함수명에 PK는 불필요해보입니다
다음 주석 내용을 추가하는건 어떨까요??
```
/* 파티션 테이블의 메타데이터 정합성을 유지하기 위해,
 * 루트 파티션 테이블에만 의미가 있는 속성들은 복사 시 제거하고,
 * 파티션에도 반드시 유지되어야 하는 제약은 그대로 보존합니다.
 *
 * EPIC CBRD-26096에 따른 전제 조건:
 *   - 파티션 키는 반드시 단일 컬럼이어야 하며,
 *   - 해당 컬럼은 PK 또는 NOT NULL UNIQUE로 정의되어야 합니다.
 *
 * 위 조건으로 인해 복합 키 여부를 별도로 검사할 필요가 없으며,
 * 단일 컬럼 단위(NOT NULL + UNIQUE) 판별만으로 파티션 키를 식별할 수 있습니다.
 *
 * 처리 규칙 요약:
 *   - PK(Primary Key)는 루트 테이블에서 정의되고 각 파티션으로 상속되므로
 *     여기에서 초기화하지 않습니다.
 *   - 반면, UNIQUE / REVERSE_UNIQUE 플래그는 루트 테이블의 메타데이터가
 *     그대로 복사되면 안 되므로, 파티션 키로 사용되는 단일 NOT NULL UNIQUE 컬럼
 *     한 개만 예외적으로 유지하고 나머지는 제거합니다.
 *   - FOREIGN KEY 및 PARTITION KEY 플래그 또한 루트 테이블 맥락 정보이므로
 *     파티션에서는 초기화합니다.
 */
```

Reply:
우선, 그대로 두시죠.